tools/Google.Cloud.Tools.ReleaseManager/NonSourceGenerator.cs

```diff
@@ -0,0 +1,689 @@
+        { "Microsoft.NET.Test.Sdk", DefaultVersionValue },
+        { "xunit", DefaultVersionValue },
+        { "xunit.runner.visualstudio", DefaultVersionValue },
+        { "Xunit.SkippableFact", DefaultVersionValue },
+        { "NSubstitute", DefaultVersionValue },
+        { "System.Linq.Async", DefaultVersionValue },
+    };
+
+    // Hard-coded versions for dependencies for production packages that can be updated arbitrarily, as their assets are all private.
+    // The relationship between this and PrivateAssets is tested on startup.
+    private static readonly Dictionary<string, string> CommonHiddenProductionDependencies = new Dictionary<string, string>
+    {
+        { ConfigureAwaitAnalyzer, DefaultVersionValue },
+    };
+
+    private const string ConfigureAwaitAnalyzer = "ConfigureAwaitChecker.Analyzer";
+    private const string CSharpWorkspacesPackage = "Microsoft.CodeAnalysis.CSharp.Workspaces";
+
+    /// <summary>
+    /// For packages which need a PrivateAssets attribute in dependencies, this dictionary provides the value of the attribute.
+    /// </summary>
+    private static readonly Dictionary<string, string> PrivateAssets = new Dictionary<string, string>
+    {
+        { GrpcCorePackage, "None" },
+        { ConfigureAwaitAnalyzer, "All" },
+        { CSharpWorkspacesPackage, "All" }
+    };
+
+    /// <summary>
+    /// Packages which are okay to keep using defaults even in patch versions, as they don't affect the released library.
+    /// </summary>
+    private static readonly HashSet<string> PermittedPatchDefaultDependencies = new() { ConfigureAwaitAnalyzer };
+
+    private static readonly IReadOnlyList<string> RenovateIgnorePaths = new List<string>
+        {
+            "issues/**"
+        }.AsReadOnly();
+
+    private readonly ApiCatalog _catalog;
+    private readonly string _generatorInputDirectory;
+    private readonly string _outputDirectory;
+
+    /// <summary>
+    /// This is in lieu of a unit test... just make sure that all the dependencies we're hard-coding the values for
+    /// really do hide all their assets.
+    /// </summary>
+    static NonSourceGenerator()
+    {
+        var brokenDependencies = CommonHiddenProductionDependencies.Keys.Where(p => !PrivateAssets.ContainsKey(p) || PrivateAssets[p] != "All").ToList();
+        if (brokenDependencies.Any())
+        {
+            throw new Exception($"ProjectGenerator error: invalid CommonHiddenProductionDependencies: {string.Join(", ", brokenDependencies)}");
+        }
+    }
+
+    // Creates an instance of the generator assuming "in-place" generation.
+    internal NonSourceGenerator(ApiCatalog catalog)
+        // TODO: Tidy this up.
+        : this(catalog, Path.Combine(DirectoryLayout.DetermineRootDirectory(), DirectoryLayout.GeneratorInput), DirectoryLayout.DetermineRootDirectory())
+    {
+    }
+
+    internal NonSourceGenerator(ApiCatalog catalog, string generatorInputDirectory, string outputDirectory)
```

Comment:
Shouldn't the ApiCatalog be obtained from the input directory here?

Or is this the ApiCatalog previously obtained from the input, but possibly augmented with entries missing for raw-generation?